### PR TITLE
Wxwidgets

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: eclass/wxwidgets.eclass [hunk 1]
```diff
@@ -24,6 +24,21 @@
...
 _WXWIDGETS_ECLASS=1
 
+# @ECLASS-VARIABLE: WX_GTK_VER
+# @PRE_INHERIT
+# @REQUIRED
+# @DESCRIPTION:
+# The SLOT of the x11-libs/wxGTK you're targeting. Needs to be defined before
+# inheriting the eclass. Can be either "3.0" or "3.0-gtk3".
+if ! declare -p WX_GTK_VER &>/dev/null; then
+	die 'WX_GTK_VER not declared'
```
Comment (thesamesam) [inline eclass/wxwidgets.eclass:34]:
```suggestion
	die "WX_GTK_VER not declared"
```

File: eclass/wxwidgets.eclass [hunk 2]
```diff
@@ -66,49 +81,31 @@ esac
 
 # @FUNCTION: setup-wxwidgets
 # @DESCRIPTION:
-# Call this in your ebuild to set up the environment for wxGTK.  Besides
-# controlling the wx-config wrapper this exports WX_CONFIG containing
-# the path to the config in case it needs to be passed to a build system.
+# Call this in your ebuild to set up the environment for wxGTK in src_configure.
+# Besides controlling the wx-config wrapper this exports WX_CONFIG containing
```
Comment (thesamesam) [inline eclass/wxwidgets.eclass:85]:
```suggestion
# Besides controlling the wx-config wrapper, this exports WX_CONFIG containing
```

File: eclass/wxwidgets.eclass [hunk 3]
```diff
@@ -66,49 +81,31 @@ esac
...
-# from the libraries.  In these versions debugging features are enabled
-# by default and need to be disabled at the package level.  Because this
-# causes many warning dialogs to pop up during runtime we add -DNDEBUG to
-# CPPFLAGS to disable debugging features (unless your ebuild has a debug
-# USE flag and it's enabled).  If you don't like this behavior you can set
-# WX_DISABLE_NDEBUG to override it.
+# This function also controls the level of debugging output from the libraries.
+# Debugging features are enabled by default and need to be disabled at the
+# package level. Because this causes many warning dialogs to pop up during
+# runtime we add -DNDEBUG to CPPFLAGS to disable debugging features (unless
```
Comment (thesamesam) [inline eclass/wxwidgets.eclass:91]:
```suggestion
# runtime, we add -DNDEBUG to CPPFLAGS to disable debugging features (unless
```

File: eclass/wxwidgets.eclass [hunk 4]
```diff
@@ -66,49 +81,31 @@ esac
...
+# your ebuild has a debug USE flag and it's enabled). If you don't like this
+# behavior you can set WX_DISABLE_NDEBUG to override it.
```
Comment (thesamesam) [inline eclass/wxwidgets.eclass:93]:
```suggestion
# behavior, you can set WX_DISABLE_NDEBUG to override it.
```

File: eclass/wxwidgets.eclass [hunk 5]
```diff
@@ -66,49 +81,31 @@ esac
...
-			wxtoolkit=gtk2
-			wxdebug="release-"
-			has_version x11-libs/wxGTK:${WX_GTK_VER}[debug] && wxdebug="debug-"
-			;;
-		*)
-			die "Invalid WX_GTK_VER: must be set to a valid wxGTK SLOT"
-			;;
+	case ${WX_GTK_VER} in
+		3.0-gtk3) wxtoolkit=gtk3 ;;
+		3.0)      wxtoolkit=gtk2 ;;
```
Comment (thesamesam) [inline eclass/wxwidgets.eclass:101]: Consider adding an eqawarn nag on the gtk2 variants? We're already filing [bugs](https://bugs.gentoo.org/618642) but this is a useful nudge.

File: eclass/wxwidgets.eclass [hunk 6]
```diff
@@ -66,49 +82,31 @@ esac
...
+		*)        die "Invalid WX_GTK_VER: must be set to a valid wxGTK SLOT" ;;
 	esac
 
+	if [[ -z ${WX_DISABLE_NDEBUG} ]]; then
+		( in_iuse debug && use debug ) || append-cppflags -DNDEBUG
```
Comment (ulm) [inline eclass/wxwidgets.eclass:104]: This will fork a subshell. Either use `{ }` or omit the parentheses entirely.

**Review by dm0- — APPROVED** [review]
Works for me.

**Pull request closed without merging**